### PR TITLE
New module to set alert email addresses for Pure Storage FlashArray

### PR DESCRIPTION
**Pull request by sdodsley**
##### SUMMARY
Add new module to add/delete/enable/disable alert email address for a Pure Storage FlashArray.
Deleting the built-in alert email `flasharray-alerts@purestorage.com` is not allowed

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
purefa_alert.yaml

### CODE REVIEW

**Comment by sdodsley** [comment]
!needs_revision

**Review by genegr — APPROVED** [review]
shipit

File: lib/ansible/modules/storage/purestorage/purefa_alert.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            array.create_alert_recipient(module.params['address'])
+            changed = True
+        except Exception:
+            module.fail_json(msg='Failed to create alert emaul: {0}'.format(module.params['address']))
+
+        if not module.params['enabled']:
+            try:
+                array.disable_alert_recipient(module.params['address'])
+            except Exception:
+                module.fail_json(msg='Failed to create alert emaul: {0}'.format(module.params['address']))
```
Comment (anshulbehl) [inline lib/ansible/modules/storage/purestorage/purefa_alert.py]: Typo, should be email

File: lib/ansible/modules/storage/purestorage/purefa_alert.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+
+def enable_alert(module, array):
+    """Enable Alert Email"""
+    changed = False
+    if not module.check_mode:
+        try:
+            array.enable_alert_recipient(module.params['address'])
+        except Exception:
+            module.fail_json(msg='Failed to enable alert emaul: {0}'.format(module.params['address']))
```
Comment (anshulbehl) [inline lib/ansible/modules/storage/purestorage/purefa_alert.py]: Same typo as above

File: lib/ansible/modules/storage/purestorage/purefa_alert.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+
+def disable_alert(module, array):
+    """Disable Alert Email"""
+    changed = False
+    if not module.check_mode:
+        try:
+            array.disable_alert_recipient(module.params['address'])
+        except Exception:
+            module.fail_json(msg='Failed to disable alert emaul: {0}'.format(module.params['address']))
```
Comment (anshulbehl) [inline lib/ansible/modules/storage/purestorage/purefa_alert.py]: Typo

File: lib/ansible/modules/storage/purestorage/purefa_alert.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+def delete_alert(module, array):
+    """Delete Alert Email"""
+    changed = False
+    if module.params['address'] == "flasharray-alerts@purestorage.com":
+        module.fail_json(msg='Built-in address {0} cannot be deleted.'.format(module.params['address']))
+    if not module.check_mode:
+        try:
+            array.delete_alert_recipient(module.params['address'])
+        except Exception:
+            module.fail_json(msg='Failed to delete alert emaul: {0}'.format(module.params['address']))
```
Comment (anshulbehl) [inline lib/ansible/modules/storage/purestorage/purefa_alert.py]: Typo

File: lib/ansible/modules/storage/purestorage/purefa_alert.py (hunk 1 above)
Comment (sdodsley) [reply lib/ansible/modules/storage/purestorage/purefa_alert.py]: Fixed

**Review by JohnLieske — APPROVED** [review]
looks good

**Pull request merged**